packages/shared/src/utils/files/file-uploader-utils.ts

```diff
@@ -65,22 +42,31 @@ export const compressImageFiles = (files: TFile[]) => {
     return Promise.all(promises);
 };
 
-export const readFiles = (files: TFile[], getFileReadErrorMessage: (t: string) => string, settings: TSettings) => {
-    const promises: Promise<TFileObject | { message: string }>[] = [];
+export const readFiles = (
+    files: Blob[],
+    getFileReadErrorMessage: (t: string) => string,
+    settings?: Partial<TSettings>
+) => {
+    const promises: Array<Promise<Partial<TFileObject> | { message: string }>> = [];
```

Comment:
just a suggestion:
maybe we could use Promise.all(files.map()) instead of using for each and adding promises inside the variable outside?

Reply:
keeping the forEach to make the file more readable